core/src/main/java/org/apache/calcite/rel/metadata/RelMdPredicates.java

```diff
@@ -578,6 +577,7 @@ public RelOptPredicateList inferPredicates(
       final Set<RexNode> allExprs = new HashSet<>(this.allExprs);
       final JoinRelType joinType = joinRel.getJoinType();
       switch (joinType) {
+      case SEMI:
```

Comment:
Shouldn't we add `case ANTI:` in all the places where `case SEMI:` was added?

Reply:
I'm not that sure, i add some `ANTI`s where `SEMI`s appears which i'm confident that the modification is right, for the others, i would rather leave it to the one who implements the `ANTI` join expression.

Reply:
Ok, I guess we can do it in a further step, thanks for the clarification